File: python/BRIDGE_WHOLE_PART/BRIDGE_WHOLE_PART/bridge/Membresia.py
from BRIDGE_WHOLE_PART.bridge.ProductoCine import ProductoCine
from .ImplementadorProductoCine import ImplementadorProductoCine
from BRIDGE_WHOLE_PART.whole_part.membresia.MembresiaSencilla import MembresiaSencilla
from BRIDGE_WHOLE_PART.whole_part.membresia.MembresiaVIP import MembresiaVIP


class Membresia(ProductoCine):
    
    implementadorProductoCine : ImplementadorProductoCine
    
    def __init__(self, tipoProductoCine, datos) -> None:
        super().__init__(tipoProductoCine)
        
        if self.tipoProductoCine == "sencilla":
            self.implementadorProductoCine = MembresiaSencilla()
        elif self.tipoProductoCine == "vip":
            self.implementadorProductoCine = MembresiaVIP()
        else: print("No existe ese producto!")
        
        self.crearProductoCine(datos)
    
    def crearProductoCine(self, datos):
        self.crearMembresia(datos)
    
    def crearMembresia(self, datos):
        self.implementadorProductoCine.completarProductoCine(datos)
    
    def getMembresia(self):
        return self.implementadorProductoCine
    
    def mostrarMembresia(self):
        self.implementadorProductoCine.mostrarProductoCine()

File: php/BRIDGE_WHOLE_PART/bridge/Membresia.php
<?php


require_once 'ProductoCine.php';
require_once './whole_part/membresia/MembresiaSencilla.php';
require_once './whole_part/membresia/MembresiaVIP.php';


class Membresia extends ProductoCine{
    
    public $implementadorProductoCine;
    
    public function __construct($tipoProductoCine) {
        parent::__construct($tipoProductoCine);
    }
    
    public function crearProducto($datos) {
        
        if ($this->tipoProductoCine == 'sencilla'){
            $this->implementadorProductoCine = new MembresiaSencilla();
        }else if($this->tipoProductoCine == 'VIP'){
            $this->implementadorProductoCine = new MembresiaVIP();
        }else{
            echo 'No existe ese producto';
        }
        
        $this->crearMembresia($datos);
    }
    
    public function crearMembresia($datos) {
        $this->implementadorProductoCine->completarProductoCine($datos);
    }
    public function getMembresia() {
        return $this->implementadorProductoCine;
    }
    
    public function mostrarMembresia() {
        return $this->implementadorProductoCine->mostrarProductoCine();
    }
}
